app/models/user.rb

```diff
@@ -0,0 +1,75 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: %i[facebook]
+
+  validates :first_name, presence: true, length: { maximum: 20 },
+  format: { with: /[a-z]+/i }
+  validates :last_name, presence: true, length: { maximum: 20 },
+  format: { with: /[a-z]+/i }
+
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  has_many :friend_requests_sent, foreign_key: :sender_id,
+          class_name: "FriendRequest", dependent: :destroy
+
+  has_many :friend_requests_received, foreign_key: :receiver_id,
+          class_name: "FriendRequest", dependent: :destroy
+
+
+  has_many :friendships, foreign_key: :adder_id, dependent: :destroy
+  has_many :friendships_2, class_name: "Friendship", foreign_key: :added_id,
+            dependent: :destroy
+
+  has_many :friends, through: :friendships, source: :added
+
+
+
+  def establish_friendship(other_user)
```

Comment:
In relation to #3. We can have a simpler way of implementing this through associations. Please see my comments above on the associations.